crates/sui-framework/sources/collectible/collectible.move

```diff
@@ -69,7 +70,8 @@ module sui::collectible {
     ): (
         Publisher,
         Display<Collectible<T>>,
-        CollectionCreatorCap<T>
+        TransferPolicyCap<Collectible<T>>,
+        CollectionCreatorCap<T>,
```

Comment:
This seems like a lot of capabilities, so I'm trying to make sure I understand why want to have each these. Let me make two suggestions, and you can tell me why they don't make sense:

- Unify `CollectionCreatorCap` and `TransferPolicyCap` (i.e., collection creator always sets the transfer policy)
- Wrap `Publisher` in a field of `CollectionCreatorCap` rather than returning it separately

I'm totally ok with keeping all of these if we can crisply explain when to use one vs another, but would prefer to keep this simple otherwise.

Reply:
There's a couple reasons for it. 

1. This way we allow building on top of it with ease giving builders full control over each aspect of the collectible. 
2. The whole thing is built for programmable transactions, and they don't support references as return values iirc, so having a full value is easier to use right away without having a "borrow" method which returns the value + hot potato to be put back later

There might be a slight concern over having all of them in the inventory, but here's how each of them can be used:

1. Publisher is kept for possible future implementation of this type (maybe register on a marketplace or somewhere else) it's easier to build 3rd party apps relying on Publisher than on CollectionCreatorCap as P is a lower level primitive and expected to be more common.

2. Display should be stored at Account or placed into a "locker" or some shared protection something. Tho maybe it's the thing we can wrap (since new updates are fetched via "update_display" event)

3. TransferPolicy has two directions by default - it either gets wrapped into a shared Royalty policy or gets immutably shared to allow free transfers

4. CollectionCreatorCap is kept as is - it allows minting

I do agree that it seems like a lot, but it really depends on how we want to use them, and, currently, wrapping them and giving a "borrow_mut" something is not the best approach due to programmable txs limitations.

Reply:
> Sounds like we might need to support TransferPolicyCap getting torn off the creator cap at some point.
This cap is directly consumed by the `royalty` module if royalties are set.

Agree on everything else, added `sui::borrow` module as well as placed the Display inside the CreatorCap.